lib/smack/SmackRep.cpp

```diff
@@ -76,8 +137,20 @@ string SmackRep::type(const llvm::Type* t) {
     return BOOL_TYPE;
   else if (isFloat(t))
     return FLOAT_TYPE;
-  else
+  else if (isInt(t)) {
+    if (BIT_VECTOR) {
+      stringstream s;
+      s << "i" << getIntSize(t);
+      return s.str();
```

Comment:
There are still plenty of places where an `"iX"` string is constructed without using `int_type(X)`; please update all occurrences, including the one inside of `type(const llvm::Type*)`. Let's write clean code and worry about optimization later if efficiency is a problem.

Furthermore, the best place to case split on whether `int` or `iX` should be used seems to be inside of `int_type(unsigned)`.

Furthermore, we should be reading `SmackOptions::BitVector` to decide, with which the `BIT_VECTOR` field is redundant, and should be removed.